percival/carrier/configuration.py

```diff
@@ -80,14 +80,14 @@ def __setattr__(self, name, value):
 
     def parameters(self):
         """return a list of parameter names"""
-        return self._parameters.keys()
+        return list(self._parameters.keys())
 
     def get_type(self, parameter):
         return self._parameters[parameter][1]
 
     def __str__(self):
         param_str = ""
-        for (name, value) in self._parameters.items():
+        for (name, value) in list(self._parameters.items()):
```

Comment:
You don't need to copy to a new list if you are just iterating the dict. You also dont need the brackets.
```suggestion
        for name, value in self._parameters.items():
```

Reply:
I think your comments are correct; I used a tool to perform the update, and that is how it came out.
Do you think I should go through simplifying the code? I would say no because i) if I change it myself, I could introduce a bug and ii) there may be a reason why the tool chose to do it like that.

Reply:
I think the reason the tool wrapped it in a list is to ensure the behaviour is the same (python2 would construct the list and return it, python3 returns an iterator), which is sensible enough. I think it is fine.